extensions-contrib/first-last-string/src/main/java/io/druid/query/aggregation/first/StringFirstAggregator.java

```diff
@@ -0,0 +1,100 @@
+/*
+ * Licensed to Metamarkets Group Inc. (Metamarkets) under one
+ * or more contributor license agreements. See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership. Metamarkets licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package io.druid.query.aggregation.first;
+
+import io.druid.collections.SerializablePair;
+import io.druid.query.aggregation.Aggregator;
+import io.druid.segment.BaseLongColumnValueSelector;
+import io.druid.segment.BaseObjectColumnValueSelector;
+
+public class StringFirstAggregator implements Aggregator
+{
+
+  private final BaseObjectColumnValueSelector valueSelector;
+  private final BaseLongColumnValueSelector timeSelector;
+  private final Integer maxStringBytes;
+
+  protected long firstTime;
+  protected String firstValue;
+
+  public StringFirstAggregator(
+      BaseLongColumnValueSelector timeSelector,
+      BaseObjectColumnValueSelector valueSelector,
+      Integer maxStringBytes
+  )
+  {
+    this.valueSelector = valueSelector;
+    this.timeSelector = timeSelector;
+    this.maxStringBytes = maxStringBytes;
+
+    firstTime = Long.MAX_VALUE;
+    firstValue = null;
+  }
+
+  @Override
+  public void aggregate()
+  {
+    long time = timeSelector.getLong();
+    if (time < firstTime) {
+      firstTime = time;
+      Object value = valueSelector.getObject();
+
+      if (value instanceof String) {
+        firstValue = (String) value;
+
+        if (firstValue.length() > maxStringBytes) {
+          firstValue = firstValue.substring(0, maxStringBytes);
+        }
+      } else if (value instanceof SerializablePair) {
+        firstValue = ((SerializablePair<Long, String>) value).rhs;
```

Comment:
I think that if the aggregator gets a SerializablePair is because it is processing an indexed druid row instead of an input message. So, the string inside the SerializablePair was checked at indexing time.

Is it correct??

Reply:
But ... I think that better we can check it on both cases